software/base/src/main/java/brooklyn/entity/basic/SoftwareProcessImpl.java

```diff
@@ -449,11 +449,14 @@ protected final void callStartHooks() {}
         configKeys.addAll(getEntityType().getConfigKeys());
 
         for (ConfigKey<?> k: configKeys) {
+            Object value = getConfig(k);
             if (PortRange.class.isAssignableFrom(k.getType())) {
                 PortRange p = (PortRange)getConfig(k);
                 if (p != null && !p.isEmpty()) ports.add(p.iterator().next());
+            } else if (value != null && PortRange.class.isAssignableFrom(value.getClass())) {
```

Comment:
Thinking about it, I'm not sure that the first if condition is needed at all. If the key's type is assignable to PortRange, then the value must be too (or null)